### PR TITLE
Add ad support for Alabaster theme

### PR DESCRIPTION
**Pull request by agjohnson**
This updates our ad insertion code to also support the Alabaster theme. The ad
is injected currently at the bottom of the sidebar, below the search box. Some
of styling was updated for this as well.

This also makes a minor change to the CSS. `div.rst-pro` was inexplicably named,
perhaps to match something in the sphinx_rtd_theme css/html? I changed this to
`div.rtd-pro`, which makes more sense. The old rules were kept in the odd case
that the old classes/promo html were used/cached somehow -- this shouldn't be
the case though.

Also, some of the dependencies were updated.

Examples:

image

image

### CODE REVIEW

**Comment by agjohnson** [comment]
This isn't going to go out just yet, so we can hold off on merging this until we've had time to write about this and received some feedback.

**Comment by ericholscher** [comment]
What is the second screenshot above? mobile view?

**Comment by agjohnson** [comment]
Below 870px or so, yes.

File: media/css/readthedocs-doc-embed.css [hunk 1]
```diff
@@ -36,38 +36,66 @@ nav.wy-nav-side {
 .bookmark-inactive {display: none;}
 
 
-/* Read the Docs promotional block, only applicable to RTD.org */
-div.wy-menu.rst-pro {
+/* Read the Docs promotional block, only applicable to RTD.org
```
Comment (ericholscher) [inline media/css/readthedocs-doc-embed.css]: Should this go in the `donate.css` and we just load it dynamically in JS?

File: readthedocs/core/static-src/core/js/doc-embed/rtd-data.js [hunk 2]
```diff
@@ -17,9 +22,9 @@ var configMethods = {
     get_theme_name: function () {
         // Crappy heuristic, but people change the theme name on us.  So we have to
         // do some duck typing.
-        if (this.theme !== 'sphinx_rtd_theme') {
+        if (this.theme !== constants.THEME_RTD) {
             if ($('div.rst-other-versions').length === 1) {
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/doc-embed/rtd-data.js]: Should we also have heuristics around alabaster, too? Seems like we'd want to handle folks renaming it as well.

File: readthedocs/core/static-src/core/js/sponsorship.js [hunk 3]
```diff
@@ -1,29 +1,40 @@
...
+        promo_class;
+    if (this.theme == constants.THEME_RTD) {
+        menu = this.get_sphinx_rtd_theme_menu();
+        promo_class = 'wy-menu';
+    }
+    else if (this.theme == constants.THEME_ALABASTER) {
+        menu = this.get_alabaster_menu();
+        promo_class = 'alabaster';
+    }
+    if (typeof(menu) != 'undefined') {
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/sponsorship.js]: Should we do something if we think we know what the theme is, but we can't find the menu?

**Pull request merged**